packages/firestore/src/remote/remote_store.ts

```diff
@@ -407,15 +426,43 @@ export class RemoteStore implements TargetMetadataProvider {
     }
 
     if (!snapshotVersion.isEqual(SnapshotVersion.min())) {
-      const lastRemoteSnapshotVersion = await this.localStore.getLastRemoteSnapshotVersion();
-      if (snapshotVersion.compareTo(lastRemoteSnapshotVersion) >= 0) {
-        // We have received a target change with a global snapshot if the snapshot
-        // version is not equal to SnapshotVersion.min().
-        await this.raiseWatchSnapshot(snapshotVersion);
+      try {
+        const lastRemoteSnapshotVersion = await this.localStore.getLastRemoteSnapshotVersion();
+        if (snapshotVersion.compareTo(lastRemoteSnapshotVersion) >= 0) {
+          // We have received a target change with a global snapshot if the snapshot
+          // version is not equal to SnapshotVersion.min().
+          await this.raiseWatchSnapshot(snapshotVersion);
+        }
+      } catch (e) {
+        logDebug(LOG_TAG, 'Failed to raise snapshot: ' + e);
+        await this.disableNetworkUntilRecovery(e);
       }
     }
   }
 
+  /**
+   * Recovery logic for IndexedDB errors that takes the network offline until
+   * IndexedDb probing succeeds. Retries are scheduled with backoff using
+   * `enqueueRetryable()`.
+   */
+  private async disableNetworkUntilRecovery(e: FirestoreError): Promise<void> {
+    if (e.name === 'IndexedDbTransactionError') {
+      ++this.indexedDbFailureBarrier;
+      await this.disableNetworkInternal();
```

Comment:
`disableNetworkInternal` does not change the `onlineStateTracker`, but it seems like while we've intentionally disabled the network we should mark ourselves offline, no?

Reply:
That will create some flicker, but it is the right thing to do. Updated.

Reply:
I had to add the state transition here since we don't raise offline events when `disableNetworkInternal` is called from `shutdown` or for primary state transitions.